src/Authentication/Authentication/Extensions/PSCmdletExtensions.cs

```diff
@@ -0,0 +1,72 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph.PowerShell.Authentication.Extensions
+{
+    using Microsoft.Graph.PowerShell.Authentication.Helpers;
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Linq.Expressions;
+    using System.Management.Automation;
+    internal static class PSCmdletExtensions
+    {
+        /// <summary>
+        /// Executes a PowerShell script.
+        /// </summary>
+        /// <typeparam name="T">The output type to return.</typeparam>
+        /// <param name="cmdlet">The executing cmdlet.</param>
+        /// <param name="contents">The PowerShell scrip to execute.</param>
+        /// <returns>The result for the executed script.</returns>
+        public static List<T> ExecuteScript<T>(this PSCmdlet cmdlet, string contents)
+        {
+            List<T> output = new List<T>();
+
+            using (PowerShell powershell = PowerShell.Create(RunspaceMode.CurrentRunspace))
+            {
+                powershell.AddScript(contents);
+                Collection<T> result = powershell.Invoke<T>();
+
+                if (cmdlet.SessionState != null)
+                {
+                    powershell.Streams.Error.ForEach(e => cmdlet.WriteError(e));
+                    powershell.Streams.Verbose.ForEach(r => cmdlet.WriteVerbose(r.Message));
+                    powershell.Streams.Warning.ForEach(r => cmdlet.WriteWarning(r.Message));
+                }
+
+                if (result != null && result.Count > 0)
+                {
+                    output.AddRange(result);
+                }
+            }
+
+            return output;
+        }
+
+        /// <summary>
+        /// Determines is a parameter has been provided by the user.
+        /// </summary>
+        /// <param name="cmdlet">The executing cmdlet.</param>
+        /// <param name="parameterName">The name of the parameter to check.</param>
+        /// <returns>True is the parameter was set by the user, otherwise false.</returns>
+        public static bool IsBound(this PSCmdlet cmdlet, string parameterName)
```

Comment:
for these two bound checking functions would it be better to go with something more like IsParameterBoundByName, IsParameterBoundByExpression.  second is a little long but the point more being that having them start with similar names  better captures they're both trying to accomplish the same thing

Reply:
I'll just overload the function `IsParameterBound` where one accepts the name of a parameter and the other an expression.